### ISSUE TITLE
Ability to disable CSV, JSON and RSS results

### ISSUE THREAD

**Issue by unixfox** [searx/searx#1610]
As a Searx public instance owner I'm getting loads of traffic from bots that abuse the CSV, JSON and RSS results. I haven't seen any human using it since I made my instance available to the public.

With a view to providing an anti bot program for the community that manage public Searx instances it would be great to have an ability to disable these CSV, JSON and RSS results in the settings because it's very hard to detect if it's a bot that does a request to that kind of results or a browser with a real human behind it due to the fact that it's not an HTML page.

Right now the only way to disable that kind of results is to modify directly the [`webapp.py`](https://github.com/asciimoo/searx/blob/master/searx/webapp.py).

**Comment by veloute** [comment]
could you use filtron for this?

**Comment by unixfox** [comment]
Maybe but I don't want to use another external tool just for that because I'm already using a custom proxy for filtering requests to Searx.
It's better to have a settings inside Searx instead of using an external tool.

**Comment by dalf** [comment]
quick hack in the source

after https://github.com/asciimoo/searx/blob/master/searx/webapp.py#L480
add
```python
	if output_format in ['csv', 'json', 'rss']:
             index_error(output_format, 'Bad Request'), 400
```

last line can be replace by ```return render('404.html'), 404``` according to the bot reactions.

**Comment by unixfox** [comment]
Yeah that's what I did a while ago, modifying the source code (like you can see [here](https://github.com/unixfox/docker-config/blob/master/searx/docker-compose.yml#L99)) but it's too hacky for me to stay with this too much time because I won't receive any update about the `webapp.py` file if someone makes a commit for that file.
